rfcs/0000-wifi-buddy-interface.md

```diff
@@ -0,0 +1,125 @@
+# RFC - ESP32 Wifi Buddy Interface Design
+
+## Goal
+
+To bring wireless networking to MnemOS Beepy devices, an [ESP32-C3] SoM will be
+integrated to serve as a WiFi coprocessor. This device will run MnemOS-based
+firmware, which we are referring to as "WiFi Buddy". See [#121] for details on
+the rationale for choosing the ESP32-C3 as the WiFi device, and for discussion
+of the overall design for ESP32-C3 integration.
+
+The goal of this document is to propose an overall design for the hardware and
+software interfaces between MnemOS on the CPU and MnemOS on the WiFi Buddy.
+
+## Background
+
+### Hardware
+
+The WiFi Buddy hardware device is a [Seeedstudio XIAO] or [Adafruit
+QT Py] ESP32-C3 development board. These are development boards for the ESP32-C3
+which are pinout compatible, so they can be used interchangeably. The ESP32-C3
+board will be treated as a system-on-module (SoM) and integrated with the MnemOS
+Beepy shield using pin headers (or potentially soldered onto the board).
+
+The following pins are broken out by the WiFi Buddy hardware and can be used for
+communication between the CPU and WiFi Buddy:
+
+- One SPI interface (`MOSI`, `MISO`, and `SCK`)
```

Comment:
There's also a chip select pin `CSn`.

Reply:
I didn't include chip select here because any of the 4 GPIOs could be the chip select pin, right?

Reply:
*sort of*. Short answer, yes it could be for now, but might be worth explicitly addressing as such.